src/config.py

```diff
@@ -34,6 +64,106 @@ def plugin_keys(cls) -> filter:
+
+        return value
+
+    @validator("logging_log_min_duration_statement")
+    @classmethod
+    def logging_log_min_duration_statement_values(cls, value: int) -> Optional[int]:
+        """Check logging_log_min_duration_statement config option is between -1 and 2147483647."""
+        if value < -1 or value > 2147483647:
+            raise ValueError("Value is not between -1 and 2147483647")
+
+        return value
+
+    @validator("memory_maintenance_work_mem")
+    @classmethod
+    def memory_maintenance_work_mem_values(cls, value: int) -> Optional[int]:
+        """Check memory_maintenance_work_mem config option is between 1024 and 2147483647."""
+        if value < 1024 or value > 2147483647:
+            raise ValueError("Value is not between 1024 and 2147483647")
+
+        return value
+
+    @validator("memory_max_prepared_transactions")
+    @classmethod
+    def memory_max_prepared_transactions_values(cls, value: int) -> Optional[int]:
+        """Check memory_max_prepared_transactions config option is between 0 and 262143."""
+        if value < 0 or value > 262143:
+            raise ValueError("Value is not between 0 and 262143")
+
+        return value
+
+    @validator("memory_shared_buffers")
+    @classmethod
+    def memory_shared_buffers_values(cls, value: int) -> Optional[int]:
+        """Check memory_shared_buffers config option is greater or equal than 16."""
+        if value < 16:
+            raise ValueError("Shared buffers config option should be at least 16")
+
+        return value
+
+    @validator("memory_temp_buffers")
+    @classmethod
+    def memory_temp_buffers_values(cls, value: int) -> Optional[int]:
+        """Check memory_temp_buffers config option is between 100 and 1073741823."""
+        if value < 100 or value > 1073741823:
+            raise ValueError("Value is not between 100 and 1073741823")
+
+        return value
+
+    @validator("memory_work_mem")
+    @classmethod
+    def memory_work_mem_values(cls, value: int) -> Optional[int]:
+        """Check memory_work_mem config option is between 64 and 2147483647."""
+        if value < 64 or value > 2147483647:
+            raise ValueError("Value is not between 64 and 2147483647")
+
+        return value
+
+    @validator("optimizer_constraint_exclusion")
+    @classmethod
+    def optimizer_constraint_exclusion_values(cls, value: str) -> Optional[str]:
+        """Check optimizer_constraint_exclusion config option is one of `on`, `off` or `partition`."""
+        if value not in ["on", "off", "partition"]:
+            raise ValueError("Value not one of 'on', 'off' or 'partition'")
+
+        return value
+
+    @validator("optimizer_default_statistics_target")
+    @classmethod
+    def optimizer_default_statistics_target_values(cls, value: int) -> Optional[int]:
+        """Check optimizer_default_statistics_target config option is between 1 and 10000."""
+        if value < 1 or value > 10000:
```

Comment:
I think we can use [conint](https://docs.pydantic.dev/1.10/usage/types/#arguments-to-conint) instead of custom validators.

Reply:
Thanks for the suggestion! I created https://github.com/canonical/postgresql-k8s-operator/issues/295 to evaluate that later.